pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -330,9 +330,11 @@ func (s *ServiceEntryStore) serviceEntryHandler(old, curr config.Config, event m
 	if event == model.EventDelete {
 		s.deleteExistingInstances(ckey, serviceInstances)
 	} else {
-		oldInstances := s.convertServiceEntryToInstances(old, cs, s.Cluster())
-		// First, delete the existing instances to avoid leaking memory.
-		s.serviceInstances.deleteInstances(ckey, oldInstances)
+		if old.Spec != nil {
```

Comment:
Is it possible for spec to be nil?

Reply:
yes, for add/delete event
https://github.com/istio/istio/blob/master/pilot/pkg/config/kube/crdclient/cache_handler.go#L55-L67

Reply:
I thought validation will catch it ?

Reply:
Not sure, i meet a panic in ut and then add the check here

Reply:
The validation works before, when a se created, we pass an empty Config(old) into the callback